### PR TITLE
Add tracing support

### PR DESCRIPTION
**Pull request by khotyn**
### Issues associated with this PR

#246 

### Sign the CLA
Make sure you has singed the [CLA](https://www.clahub.com/agreements/alipay/sofa-mosn)

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, 
details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is need if code is changed, your unit test should cover boundary cases, corner cases and some exception cases.
And you need to show the ut result.

### Benchmark
If your code involve the processing of every request, you should give the Benchmark Result

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

**Comment by khotyn** [comment]
This PR is still working in process.

**Comment by khotyn** [comment]
HTTP support should follow the w3c specification: 

* https://w3c.github.io/distributed-tracing/report-trace-context.html
* https://w3c.github.io/distributed-tracing/report-correlation-context.html

**Comment by nejisama** [comment]
整体的部分，除了冲突需要解决以外，有几个地方也需要注意一下
1.  code style 需要按照goimports来处理一下，import的分组； 新增的文件需要加license
2. 代码里的中文注释去掉
3. 新增了依赖opentracing，但是由于现在mosn的vendor管理没弄好，dep ensure把其他的都更新了，这导致PR看上去改动的内容很多，可以看下把本次PR无关的vendor 更新去掉，这里会在12月迭代统一做一个vendor的管理；之后就可以用dep ensure了

File: pkg/trace/sofatracer.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+type SofaTracer struct {
+	spanChan      chan *SofaTracerSpan
+	ingressLogger log.Logger
+	egressLogger  log.Logger
+}
+
+func newSofaTracer() *SofaTracer {
+	instance := &SofaTracer{}
+	instance.spanChan = make(chan *SofaTracerSpan)
+
```
Comment (nejisama) [inline pkg/trace/sofatracer.go]: 这个channel 没有buffer， 写的时候可能会造成block

File: pkg/proxy/downstream.go [hunk 2]
```diff
@@ -725,6 +735,32 @@ func (s *downStream) onUpstreamData(data types.IoBuffer, endStream bool) {
 	}
 
 	s.appendData(data, endStream)
+	s.finishTracing(strconv.Itoa(types.SuccessCode))
```
Comment (nejisama) [inline pkg/proxy/downstream.go]: finishTracing的调用，考虑应该做在onUpstreamResponseRecvFinished里
